database-commons/src/main/java/io/cdap/plugin/db/batch/sink/AbstractDBSink.java

```diff
@@ -327,15 +333,21 @@ private void validateFields(Schema inputSchema, ResultSet rs) throws SQLExceptio
       boolean isColumnNullable = (ResultSetMetaData.columnNullable == rsMetaData.isNullable(columnIndex));
 
       Schema.Type columnType = columnSchema.getType();
-      Schema.Type fieldType = field.getSchema().isNullable() ? field.getSchema().getNonNullable().getType()
-        : field.getSchema().getType();
+      Schema.LogicalType columnLogicalType = columnSchema.getLogicalType();
+
+      Schema fieldSchema = field.getSchema().isNullable() ? field.getSchema().getNonNullable() : field.getSchema();
+      Schema.Type fieldType = fieldSchema.getType();
+      Schema.LogicalType fieldLogicalType = fieldSchema.getLogicalType();
+
       boolean isNotNullAssignable = !isColumnNullable && field.getSchema().isNullable();
-      boolean isNotCompatible = !Objects.equals(fieldType, columnType);
+      boolean isNotCompatible = !(Objects.equals(fieldType, columnType)
+        && Objects.equals(fieldLogicalType, columnLogicalType));
 
       if (isNotCompatible) {
         invalidFields.add(field.getName());
-        LOG.error("Field {} was given as type {} but the database column is actually of type {}.", field.getName(),
-                  fieldType, columnSchema.getType());
+        LOG.error("Field {} was given as type {} and logical type {} " +
```

Comment:
users don't actually know anything about physical and logical types. They just see a single type in the UI, where timestamp is in the same dropdown as long.

Instead of logging both, we should keep the old message, but use the LogicalType.token() value if the logical type is not null and the type otherwise.

Reply:
Done